File: 1.1/src/cz/mg/vulkan/PFNvkTrimCommandPoolKHR.java
package cz.mg.vulkan;

public class PFNvkTrimCommandPoolKHR extends VkFunctionPointer {
    public PFNvkTrimCommandPoolKHR() {
    }

    protected PFNvkTrimCommandPoolKHR(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkTrimCommandPoolKHR(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkTrimCommandPoolKHR(long value) {
        setValue(value);
    }

    public PFNvkTrimCommandPoolKHR(VkInstance instance) {
        super(instance, new VkString("vkTrimCommandPoolKHR"));
    }

    public void call(VkDevice device, VkCommandPool commandPool, VkCommandPoolTrimFlags flags){
        call(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddressNative(), commandPool != null ? commandPool.getVkAddress() : VkPointer.getNullAddressNative(), flags != null ? flags.getVkAddress() : VkPointer.getNullAddressNative());
    }

    protected static native void call(long vkaddress, long device, long commandPool, long flags);
}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkTrimCommandPoolKHR.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkTrimCommandPoolKHR_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong commandPool, jlong flags){
    (void)env;
    (void)jc;
    PFN_vkTrimCommandPoolKHR f = (PFN_vkTrimCommandPoolKHR)jniLongToFunctionPointer(address);
    f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkCommandPool*)jniLongToPointer(commandPool)),
        *((VkCommandPoolTrimFlags*)jniLongToPointer(flags))
    );
}


void Java_cz_mg_vulkan_PFNvkTrimCommandPoolKHR_callSimplifiedNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong commandPool, jint flags){
    (void)env;
    (void)jc;
    PFN_vkTrimCommandPoolKHR f = (PFN_vkTrimCommandPoolKHR)jniLongToFunctionPointer(address);
    f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkCommandPool*)jniLongToPointer(commandPool)),
        (VkCommandPoolTrimFlags)flags
    );
}

